uniffi_bindgen/src/interface/mod.rs

```diff
@@ -657,12 +657,33 @@ impl ComponentInterface {
     /// The set of FFI functions is derived automatically from the set of higher-level types
     /// along with the builtin FFI helper functions.
     pub fn iter_ffi_function_definitions(&self) -> impl Iterator<Item = FfiFunction> + '_ {
-        self.iter_user_ffi_function_definitions()
+        self.iter_ffi_function_definitions_conditionally_include_checksums(true)
+    }
+
+    pub fn iter_ffi_function_definitions_excluding_checksums(
+        &self,
+    ) -> impl Iterator<Item = FfiFunction> + '_ {
+        self.iter_ffi_function_definitions_conditionally_include_checksums(false)
+    }
+
+    fn iter_ffi_function_definitions_conditionally_include_checksums(
+        &self,
+        include_checksums: bool,
+    ) -> impl Iterator<Item = FfiFunction> + '_ {
+        let iterator = self
+            .iter_user_ffi_function_definitions()
             .cloned()
             .chain(self.iter_rust_buffer_ffi_function_definitions())
             .chain(self.iter_futures_ffi_function_definitions())
-            .chain(self.iter_checksum_ffi_functions())
-            .chain([self.ffi_uniffi_contract_version()])
+            .chain([self.ffi_uniffi_contract_version()]);
```

Comment:
this change puts `ffi_uniffi_contract_version ` **before** in `iter_checksum_ffi_functions`, but it should not have any implications is code. Since it just affects the order in where all generated source code places the `uniffiCheckContractApiVersion` and `checksum` functions. Which does not incur any logic change.